datafusion/expr/src/expr_rewriter/order_by.rs

```diff
@@ -136,24 +162,24 @@ mod test {
 
         let cases = vec![
             TestCase {
-                desc: "c1 --> t.c1",
```

Comment:
I am not sure why direct aggregate creation doesn't need this rewrite anymore. All the other tests in DataFusion CI pass and I wrote the unit tests last week to document the existing behavior in https://github.com/apache/arrow-datafusion/pull/5088

If reviewers prefer to keep the old behavior here, that is easy as the first commit in this PR,  https://github.com/apache/arrow-datafusion/pull/5067/commits/0c16ef1d0965e4f59a3bb6880f225d7ddc56b47d actually keeps all these tests passing with the existing "aggregate" code.